testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/saml/SamlBackchannelLogoutReceiver.java

```diff
@@ -0,0 +1,90 @@
+package org.keycloak.testsuite.util.saml;
+
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import org.keycloak.dom.saml.v2.protocol.LogoutRequestType;
+import org.keycloak.dom.saml.v2.protocol.StatusResponseType;
+import org.keycloak.protocol.saml.profile.util.Soap;
+import org.keycloak.saml.SAML2LogoutResponseBuilder;
+import org.keycloak.saml.processing.api.saml.v2.response.SAML2Response;
+import org.keycloak.saml.processing.core.saml.v2.common.SAMLDocumentHolder;
+import org.w3c.dom.Document;
+
+import javax.ws.rs.core.HttpHeaders;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+
+public class SamlBackchannelLogoutReceiver implements AutoCloseable {
+
+    private final HttpServer server;
+    private LogoutRequestType logoutRequest;
+    private final String url;
+    private final String clientEntityId;
+
+    public SamlBackchannelLogoutReceiver(int port, String clientEntityId) {
+        this.clientEntityId = clientEntityId;
+        try {
+            InetSocketAddress address = new InetSocketAddress(InetAddress.getByName("localhost"), port);
+            server = HttpServer.create(address, 0);
+            this.url = "http://" + address.getHostString() + ":" + port;
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot create http server", e);
+        }
+
+        server.createContext("/", new SamlBackchannelLogoutReceiver.SamlBackchannelLogoutHandler());
+        server.setExecutor(null);
+        server.start();
+    }
+
+    public String getUrl() {
+        return url;
+    }
+
+    public boolean isLogoutRequestReceived() {
+        return logoutRequest != null;
+    }
+
+    public LogoutRequestType getLogoutRequest() {
+        return logoutRequest;
+    }
+
+    @Override
+    public void close() throws Exception {
+        server.stop(0);
+    }
+
+    private class SamlBackchannelLogoutHandler implements HttpHandler {
+        public void handle(HttpExchange t) throws IOException {
+            try {
+                t.getResponseHeaders().add(HttpHeaders.CONTENT_TYPE, "text/xml");
+                t.sendResponseHeaders(500, 0);
```

Comment:
Indeed, it was a typo --> changed to 200. It did not make the test fail, since the library sending the SOAP request accepts 500 as valid response, since SOAP-Faults are returned with 500. It is then the responsibility of the client to parse the body (the SOAP Fault).